core/src/main/java/org/elasticsearch/search/internal/DefaultSearchContext.java

```diff
@@ -564,6 +567,11 @@ public FieldDoc searchAfter() {
         return searchAfter;
     }
 
+    public SearchContext sliceFilter(SliceQuery filter) {
```

Comment:
instead of adding this setter can we pass a filter as a ctor arg to the `DefaultSearchContext` and just wrap the filter or replace it that comes from `public Query searchFilter(String[] types)` instead of changing the logic here?